### ISSUE TITLE
ECS: Remove forEach on queries

### ISSUE THREAD

**Issue by christopher-buss** [AetherInteractiveLtd/Tina#42]
ForEach is useful as sadly we have the inability (yet!) to add custom iterators to objects.

We do though have a `.iter()` method that basically achieves the same functionality (albeit is not as pretty as direct iteration). The `.iter()` method has less overhead due to not needing to call a function for every iteration, while also being more consistent with regular loops (we don't have to use return false to break out of a loop!).

**Comment by RigidStudios** [comment]
Yeah I'll vouch for this, open a PR when you feel like you want to do it (the earlier the better). Same one that satisfies #47 preferrably.
